### PR TITLE
reducing is_topic_type idl due to out of memory errors

### PR DESCRIPTION
**Pull request by ClaytonCalabrese**

### CODE REVIEW

File: tests/DCPS/Compiler/is_topic_type/more_is_topic_type.idl [hunk 1]
```diff
@@ -0,0 +1,80 @@
+@default_nested
```
Comment (iguessthislldo) [inline tests/DCPS/Compiler/is_topic_type/more_is_topic_type.idl:1]: Using `@default_nested` on the modules is the common factor, how about naming the file after that?

File: tests/DCPS/Compiler/is_topic_type/run_test.pl [hunk 2]
```diff
@@ -28,6 +28,9 @@
   "NonAnnotatedModule/NestedStruct"                => {not_found => 1, dn => 0, no_dn => 0},
   "NonAnnotatedModule/NestedTrueStruct"            => {not_found => 1, dn => 0, no_dn => 0},
   "NonAnnotatedModule/NestedFalseStruct"           => {not_found => 1, dn => 1, no_dn => 1},
+);
+
+my %expected2 = (
```
Comment (iguessthislldo) [inline tests/DCPS/Compiler/is_topic_type/run_test.pl:33]: How about keeping the hashes combined and record the file name to check in `subtest`?

File: tests/DCPS/Compiler/is_topic_type/run_test.pl [hunk 3]
```diff
@@ -76,19 +79,47 @@ sub subtest {
...
     }
   }
-  # Make sure we found everything in %expected
-  while (my ($k, $v) = each %expected) {
-    if ($v->{not_found}) {
-      print STDERR "ERROR in $mode: $k was not found in ITL\n";
+  return $status;
+}
+
+sub subtest2 {
```
Comment (iguessthislldo) [inline tests/DCPS/Compiler/is_topic_type/run_test.pl:85]: Copying the function is overkill. Have it take the filename as an argument and combine the hashes like I suggest above.

File: tests/DCPS/Compiler/is_topic_type/run_test.pl [hunk 4]
```diff
@@ -76,19 +79,47 @@ sub subtest {
       $status = 1;
     }
   }
-  # Make sure we found everything in %expected
```
Comment (iguessthislldo) [inline tests/DCPS/Compiler/is_topic_type/run_test.pl:79]: I think this is an important check and would like to see it stay in the test, but limit it to the filename being checked.

**Review by iguessthislldo — CHANGES_REQUESTED** [review]

File: tests/DCPS/Compiler/is_topic_type/run_test.pl (hunk 4 above)
Comment (ClaytonCalabrese) [reply tests/DCPS/Compiler/is_topic_type/run_test.pl:79]: Honestly not sure how that got deleted

File: tests/DCPS/Compiler/is_topic_type/run_test.pl [hunk 5]
```diff
@@ -78,17 +79,19 @@ sub subtest {
   }
   # Make sure we found everything in %expected
   while (my ($k, $v) = each %expected) {
-    if ($v->{not_found}) {
+    if ($v->{not_found} && $v->{f} eq $fname) {
       print STDERR "ERROR in $mode: $k was not found in ITL\n";
-      $status = 1;
+      $status = 1
```
Comment (iguessthislldo) [inline tests/DCPS/Compiler/is_topic_type/run_test.pl:84]:
```suggestion
      $status = 1;
```

**Review by iguessthislldo — CHANGES_REQUESTED** [review]

**Review by iguessthislldo — APPROVED** [review]

**Pull request merged**